examples/testhooks/delete-release.sh

```diff
@@ -5,5 +5,7 @@ set -aeuo pipefail
 # Note(turkenh): This is a workaround for the infamous dependency problem during deletion.
 # Note(ytsarev): In addition to helm Release deletion we also need to pause
 # XService reconciler to prevent it from recreating the Release.
-${KUBECTL} annotate xservices.aws.platformref.upbound.io --all crossplane.io/paused="true"
+${KUBECTL} annotate --overwrite xservices.aws.platformref.upbound.io --all crossplane.io/paused="true" || true
+${KUBECTL} annotate --overwrite xapps.aws.platformref.upbound.io --all crossplane.io/paused="true" || true
```

Comment:
Could you elaborate on this change in the test hook structure?

Reply:
`XIRSA` resource is now part of `xservice` composition  check: https://github.com/upbound/platform-ref-aws/pull/110/files#diff-9bb16b7aeaf3cd540556c7427b1a9a4d5134ea920e70d18c3dc60d44cc8da2b1R42 and it turns out in some CI runs that everytime  ` release.helm.crossplane.io "platform-ref-aws-ghost-xxxxxxx" ` was a blocker for successful deletion 

https://github.com/upbound/platform-ref-aws/actions/runs/5574439234/jobs/10183094511
https://github.com/upbound/platform-ref-aws/actions/runs/5575025366/jobs/10184400793

interesting here was that this release was reinstalled during the eks cluster entered the deletion phase - so it was stuck forever because the helm-controller was unable to find the connection secret - my idea was that the following line causes the reinstallation: https://github.com/upbound/platform-ref-aws/blob/main/examples/testhooks/delete-release.sh#L9 for that reason i changed the hook 

did you think i need to rethink this idea ?

Reply:
hmm, I've noticed the Release recreation before.

That's why we are trying to set the pause annotation first to avoid the Release recreation after the hook-based forceful deletion. 

It was one reason for the deletion hook split, so xservices and xapps are not messing with each other state.

https://github.com/upbound/platform-ref-aws/blob/main/examples/testhooks/delete-release.sh#L9 should be indeed changed to targeted deletion e.g. with the label selector so it will not interfere with xapps, but I would prefer to keep the deletion hooks in isolation. Also `|| true` does not look reliable setting of the pause annotation

In addition to that, maybe we should set the pause annotation also on the claim(not only XR) here https://github.com/upbound/platform-ref-aws/pull/110/files#diff-4d03a41a551c931d370450112f0eee4dc5d2935c4c8d0c4fcd1ca7cc8f9ad0f8L8